### PR TITLE
👯‍♀️ Improve the `plural` logging function

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

File: packages/myst-cli-utils/src/utils.ts [hunk 1]
```diff
@@ -60,5 +62,8 @@ export function plural(f: string, count?: number | any[] | Record<any, any>): st
       : Array.isArray(count)
       ? count?.length
       : Object.keys(count ?? {}).length) ?? 0;
-  return f.replace('%s', String(num)).replace(/\(s\)/g, num === 1 ? '' : 's');
+  return f
+    .replace('%s', String(num))
+    .replace(/\(s\)/g, num === 1 ? '' : 's')
+    .replace(/\(([a-z0-9A-Z-]*)\|([a-z0-9A-Z-]*)\)/g, num === 1 ? '$1' : '$2');
```
Comment (fwkoch) [inline packages/myst-cli-utils/src/utils.ts:68]: You can replace line 67 and 68 with this:

```suggestion
    .replace(/\((([a-z0-9A-Z-]*)\|)?([a-z0-9A-Z-]*)\)/g, num === 1 ? '$2' : '$3');
```

In addition to only being one regex, it handles additional cases that were not covered before, e.g. `stitch(es)`.

File: packages/myst-cli-utils/src/utils.ts [hunk 2]
```diff
@@ -60,5 +62,8 @@ export function plural(f: string, count?: number | any[] | Record<any, any>): st
       : Array.isArray(count)
       ? count?.length
       : Object.keys(count ?? {}).length) ?? 0;
-  return f.replace('%s', String(num)).replace(/\(s\)/g, num === 1 ? '' : 's');
+  return f
+    .replace('%s', String(num))
+    .replace(/\(s\)/g, num === 1 ? '' : 's')
```
Comment (fwkoch) [inline packages/myst-cli-utils/src/utils.ts:67]:
```suggestion
```

File: packages/myst-cli-utils/src/utils.ts (hunk 1 above)
Comment (fwkoch) [reply packages/myst-cli-utils/src/utils.ts:68]: (Would be good to chuck some tests on this though...)

Comment (rowanc1) [reply packages/myst-cli-utils/src/utils.ts:68]: Time for some tests, yeah. :)

Comment (rowanc1) [reply packages/myst-cli-utils/src/utils.ts]: Brought in your changes and added tests!

**Review by fwkoch — APPROVED** [review]

**Pull request merged**